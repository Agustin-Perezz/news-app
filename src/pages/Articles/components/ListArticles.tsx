import React from 'react';
import { Box, Grid, Typography } from '@mui/material';
import { NewsProps } from '../../../types/news-props';
import { Article } from './Article';
import { RenderController } from '../../../components/ui/RenderController';

interface Props {
  dataArticles?: NewsProps[];
  isLoading: boolean;
  isError?: string;
}

export const ListArticles: React.FC<Props> = ({
  dataArticles,
  isLoading,
  isError,
}) => {
  return (
    <RenderController
      isLoading={isLoading}
      isError={isError}
      data={dataArticles}
    >
      <Box sx={{ flexGrow: 1, my: 3 }}>
        <Typography variant="h5" mb={3} textAlign="center" color="snow">
          Results for search:
        </Typography>
        <Grid
          container
          spacing={{ xs: 2, md: 3 }}
          columns={{ xs: 0, sm: 8, md: 13 }}
          justifyContent="center"
          sx={{
            maxWidth: '1200px',
            margin: '0px auto 20px auto !important',
          }}
        >
          {dataArticles?.map((article) => (
            <Grid item xs={2} sm={4} md={3.5} key={article.uuid}>
              <Article dataArticle={article} />
            </Grid>
          ))}
        </Grid>
      </Box>
    </RenderController>
  );
};
