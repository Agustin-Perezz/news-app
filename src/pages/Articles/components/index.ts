export * from './Article';
export * from './ListArticles';
