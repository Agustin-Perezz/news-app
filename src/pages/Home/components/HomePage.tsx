import { Welcome } from '.';

const HomePage: React.FC = () => {
  return <Welcome />;
};

export default HomePage;
