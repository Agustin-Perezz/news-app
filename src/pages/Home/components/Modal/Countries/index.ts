export * from './ListCountries';
export * from './dataCountries';
