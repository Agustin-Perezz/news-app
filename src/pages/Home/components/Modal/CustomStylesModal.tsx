import { styled } from '@mui/material/styles';
import MuiBox from '@mui/material/Box';
import MuiCloseIcon from '@mui/icons-material/Close';

export const CustomModalBox = styled(MuiBox)(() => ({
  position: 'absolute',
  top: '50%',
  left: '50%',
  transform: 'translate(-50%, -50%)',
  width: '400px',
  bgcolor: '#1c232c',
  boxShadow: '24',
  borderRadius: '8px',
  border: '1px solid #454663',
  padding: '12px 10px',
  backgroundColor: '#1c1c23',
}));

export const CustomCloseIcon = styled(MuiCloseIcon)(() => ({
  fill: '#887e7e',
  padding: '2px',
  background: '#3f3f3f',
  borderRadius: '50%',
  '&:hover': { fill: '#a54b4b', cursor: 'pointer', border: '1px solid #826D6D' },
  position: 'absolute',
  top: '15px',
  right: '25px',
  height: '28px',
  width: '28px',
}));
