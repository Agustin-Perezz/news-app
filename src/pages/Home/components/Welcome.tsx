import { Box, Typography } from '@mui/material';
import { useContext } from 'react';
import { ListCards } from '../../../components/ui/Cards';
import { NewsContext } from '../../../context/NewsContext';

export const Welcome: React.FC = () => {
  const { data, isLoading, isError } = useContext(NewsContext);
  return (
    <Box
      sx={{
        marginTop: '30px',
        display: 'flex',
        justifyContent: 'center',
        flexWrap: 'wrap',
      }}
    >
      <Typography
        sx={{ marginBottom: '20px', width: '100%', textAlign: 'center' }}
        variant="h4"
      >
        Welcome.
      </Typography>
      <ListCards
        dataArticles={data}
        isLoading={isLoading}
        isError={isError}
        titleText="More recent top news."
      />
    </Box>
  );
};
