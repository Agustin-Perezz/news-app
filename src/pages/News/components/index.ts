export * from '../../../components/ui/Cards/GeneralCard';
export * from '../../../components/ui/Cards/ListCards';
