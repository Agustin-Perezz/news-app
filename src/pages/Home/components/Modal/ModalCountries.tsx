import { Fade, Modal, Typography, Divider, Backdrop } from '@mui/material';
import { CustomCloseIcon, CustomModalBox } from './CustomStylesModal';
import { ListCountries } from './Countries';
import { useModalController } from '../../hooks';

export const ModalCountries: React.FC = () => {
  const { open, onClose, handleClickCountry } = useModalController();
  return (
    <Modal
      open={open}
      onClose={onClose}
      closeAfterTransition
      BackdropComponent={Backdrop}
      BackdropProps={{
        timeout: 500,
      }}
      sx={{ backgroundColor: 'rgb(0 0 0 / 35%)' }}
    >
      <Fade in={open}>
        <CustomModalBox>
          <Typography variant="h6" component="h2" pl={2}>
            Select your country.
          </Typography>
          <Divider sx={{ margin: '10px 9px 10px 12px' }} />
          <ListCountries handleClickCountry={handleClickCountry} />
          <CustomCloseIcon onClick={onClose} />
        </CustomModalBox>
      </Fade>
    </Modal>
  );
};
