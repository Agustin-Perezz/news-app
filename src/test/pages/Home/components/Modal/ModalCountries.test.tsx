import { fireEvent, render, screen } from '@testing-library/react';
import { ModalCountries } from '../../../../../pages/Home/components/Modal';
import { renderWithCache } from '../../../../utils';

type Props = { cacheValue: Record<string, string> };

describe('Test in <ModalCountries.test />', () => {
  function setup({ cacheValue }: Props) {
    renderWithCache({ cacheValue, route: '/home', children: <ModalCountries /> });
  }

  it('should match snapshot', () => {
    setup({ cacheValue: {} });
    expect(screen).toMatchSnapshot();
  });
  it('should show header title', () => {
    setup({ cacheValue: {} });
    screen.getByText('Select your country.');
  });
  it('should not show the modal if cache contain the country key', () => {
    setup({ cacheValue: { country: 'Argentina' } });
    expect(screen.queryByTestId('modal')).not.toBeInTheDocument();
  });
});
