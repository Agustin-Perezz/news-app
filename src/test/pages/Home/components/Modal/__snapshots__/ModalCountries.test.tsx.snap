// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <ModalCountries /> should match snapshot 1`] = `
<div
  aria-hidden="true"
/>
`;
